contracts/funds/Fund.sol

```diff
@@ -195,73 +196,41 @@ contract Fund is IFund, ReentrancyGuard, IERC721Receiver, Initializable {
     }
 
     function takeAction(
+        Trigger calldata trigger,
         Action calldata action,
-        ActionRuntimeParams calldata runtimeParams,
+        uint256[] calldata collaterals,
         uint256[] calldata fees
-    ) public nonReentrant onlyDeployedFund onlyFundManager returns (ActionResponse memory) {
-        return _takeAction(action, runtimeParams, fees);
+    ) public nonReentrant onlyDeployedFund onlyFundManager {
+        return _takeAction(trigger, action, collaterals, fees);
     }
 
     function takeActionToClosePosition(
+        Trigger calldata trigger,
         Action calldata action,
-        ActionRuntimeParams calldata runtimeParams,
+        uint256[] calldata collaterals,
         uint256[] calldata fees
-    ) public nonReentrant onlyDeployedFund returns (ActionResponse memory resp) {
-        require(actionPositionsMap[keccak256(abi.encode(action))].length > 0);
-        resp = _takeAction(action, runtimeParams, fees);
-        require(resp.position.nextActions.length == 0);
+    ) public nonReentrant onlyDeployedFund {
+        require(block.timestamp >= subStuff.constraints.lockin); // fund expired
+        require(roboCop.actionClosesPendingPosition(action)); // but positions are still open that can be closed by this
+        _takeAction(trigger, action, collaterals, fees);
+        // TODO: how to make sure new action does not spawn another position?
```

Comment:
why do we need to make sure new action does not spawn another position?

Reply:
If we don't make sure of this, can be exploited to keep taking more and more actions  by non-fund-manager if fundManger does not close before `lockin`. right?